compare-oracle-to-fallback/src/agent.js

```diff
@@ -1,99 +1,127 @@
+    alertId: `${developerAbbrev}-${protocolAbbrev}-FALLBACK-ORACLE-DISPARITY`,
+    severity: FindingSeverity.High,
+    type: FindingType.Degraded,
+    metadata: {
+      symbol: reserveToken.symbol,
+      tokenAddress: reserveToken.tokenAddress,
+      tokenPrice,
+      tokenPriceFallback,
+      percentError,
+      numAlertsInLastDay,
+    },
+  });
+}
 
-    // get the block timestamp
-    const blockTimestamp = new BigNumber(blockEvent.block.timestamp);
+function calculatePercentError(first, second) {
+  const delta = first.minus(second).absoluteValue();
+  return delta.div(first).multipliedBy(100);
+}
 
-    // define the promise function to run for each reserve token
-    async function checkOracleAndFallback(tokenContractFallbackAlert) {
-      const {
+// define the promise function to run for each reserve token
+async function checkOracleAndFallback(tokenContractFallbackAlert, override, blockTimestamp, data) {
+  const {
+    reserveToken,
+    aaveOracleContractInstance,
+    fallbackOracleContractInstance,
+    tokenAlert,
+  } = tokenContractFallbackAlert;
+
+  // get the asset price from the oracle
+  const oraclePriceEthers = await aaveOracleContractInstance.getAssetPrice(
+    reserveToken.tokenAddress,
+    { ...override },
+  );
+
+  // convert the ethers BigNumber type to BigNumber.js type
+  const oraclePrice = new BigNumber(oraclePriceEthers.toString());
+
+  // get the asset price from the fallback oracle
+  const fallbackPriceEthers = await fallbackOracleContractInstance.getAssetPrice(
+    reserveToken.tokenAddress,
+    { ...override },
+  );
+
+  // convert the returned value to a BigNumber type
+  const fallbackPrice = new BigNumber(fallbackPriceEthers.toString());
+
+  // calculate the percent error between the price oracle and the fallback oracle
+  const percentError = calculatePercentError(oraclePrice, fallbackPrice);
+
+  // check if the value exceeds the threshold
+  if (percentError.isGreaterThan(AAVE_ORACLE_PERCENT_ERROR_THRESHOLD)) {
+    // if less than 24 hours have elapsed, just increment the counter for the number of alerts
+    // that would have been generated
+    if (blockTimestamp.minus(tokenAlert.tStart) < ALERT_MINIMUM_INTERVAL_SECONDS) {
+      tokenAlert.numAlertsInLastDay += 1;
+    } else {
+      // restart the alert counter for this token
+      tokenAlert.numAlertsInLastDay = 0;
+      tokenAlert.tStart = new BigNumber(blockTimestamp.toString());
+
+      // create the alert
+      return createAlert(
         reserveToken,
-        aaveOracleContractInstance,
-        fallbackOracleContractInstance,
+        oraclePrice,
+        fallbackPrice,
+        percentError,
         tokenAlert,
-      } = tokenContractFallbackAlert;
-
-      // get the asset price from the oracle
-      const oraclePriceEthers = await aaveOracleContractInstance.getAssetPrice(
-        reserveToken.tokenAddress,
-        { ...override },
+        data,
       );
```

Comment:
We can't reset the counter until the finding is created (the finding contains the counter value).
```suggestion
      // create the alert
      const finding = createAlert(
        reserveToken,
        oraclePrice,
        fallbackPrice,
        percentError,
        tokenAlert,
        data,
      );
      
      // restart the alert counter for this token
      tokenAlert.numAlertsInLastDay = 0;
      tokenAlert.tStart = new BigNumber(blockTimestamp.toString());
      
      return finding;
```